### PR TITLE
Add support for TCP Client

### PR DESCRIPTION
**Pull request by khoi**

### CODE REVIEW

File: Sources/Transport/TCP/TCPTransport.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+public class TCPTransport: NSObject, Transport {
+    let host: String
+    let port: UInt16
+
+    private var completionBlocks = [Int: TransportCompletionCallback]()
+    private var tag: Int = 0
+
+    private lazy var socket: GCDAsyncSocket = {
+        let socket = GCDAsyncSocket(delegate: self,
```
Comment (NghiaTranUIT) [inline Sources/Transport/TCP/TCPTransport.swift]: I suggest returning GCDAsyncSocket's instance intermediately. Don't need to `let socket = `

Comment (khoi) [reply Sources/Transport/TCP/TCPTransport.swift]: ah true,

File: Sources/Transport/TCP/TCPTransport.swift [hunk 2]
```diff
@@ -15,16 +15,18 @@ public class TCPTransport: NSObject, Transport {
...
+    private var timeOut: TimeInterval
 
     private lazy var socket: GCDAsyncSocket = {
         let socket = GCDAsyncSocket(delegate: self,
                                     delegateQueue: DispatchQueue(label: "TCPClient_Delegate_Queue"))
         return socket
     }()
 
-    public init(host: String, port: UInt16) {
+    public init(host: String, port: UInt16, timeOut: TimeInterval = 15) {
```
Comment (NghiaTranUIT) [inline Sources/Transport/TCP/TCPTransport.swift]: There are too many arguments in Initialization. I strongly suggest we should categorize it into the structure.

I could be
```
protocol TransportConfigable {
    var host: String { get }
    var port: UInt16 { get }
    var timeOut: TimeInterval { get }
}
```
We could re-use it in UDP.

Comment (khoi) [reply Sources/Transport/TCP/TCPTransport.swift]: Doing so would complicate the public interface, remember that both `UDPTransport` and `TCPTransport` are user-facing classes. 
3 is an acceptable number, consider one is optional.

Comment (khoi) [reply Sources/Transport/TCP/TCPTransport.swift]: But it's a valid point, we'll consider moving to a configuration struct if the params grow

**Pull request merged**